packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/SummaryController.java

```diff
@@ -0,0 +1,235 @@
+						content =
+								@Content(
+										mediaType = MediaType.APPLICATION_JSON_VALUE,
+										schema =
+												@io.swagger.v3.oas.annotations.media.Schema(
+														implementation = Summary.class))),
+				@ApiResponse(responseCode = "204", description = "There was no summary found", content = @Content),
+				@ApiResponse(
+						responseCode = "500",
+						description = "There was an issue retrieving the summary from the data store",
+						content = @Content)
+			})
+	public ResponseEntity<Summary> getSummary(
+			@PathVariable("id") final UUID id,
+			@RequestParam(name = "project-id", required = false) final UUID projectId) {
+		final Schema.Permission permission =
+				projectService.checkPermissionCanRead(currentUserService.get().getId(), projectId);
+
+		final Optional<Summary> summary = summaryService.getAsset(id, permission);
+		return summary.map(ResponseEntity::ok)
+				.orElseGet(() -> ResponseEntity.noContent().build());
+	}
+
+	@PostMapping
+	@Secured(Roles.USER)
+	@Operation(summary = "Create a new summary")
+	@ApiResponses(
+			value = {
+				@ApiResponse(
+						responseCode = "201",
+						description = "Summary created.",
+						content =
+								@Content(
+										mediaType = MediaType.APPLICATION_JSON_VALUE,
+										schema =
+												@io.swagger.v3.oas.annotations.media.Schema(
+														implementation = Summary.class))),
+				@ApiResponse(
+						responseCode = "500",
+						description = "There was an issue creating the summary",
+						content = @Content)
+			})
+	public ResponseEntity<Summary> createSummary(
+			@RequestBody final Summary item,
+			@RequestParam(name = "project-id", required = false) final UUID projectId) {
+		final Schema.Permission permission =
+				projectService.checkPermissionCanWrite(currentUserService.get().getId(), projectId);
+		try {
+			return ResponseEntity.status(HttpStatus.CREATED).body(summaryService.createAsset(item, permission));
+		} catch (final IOException e) {
+			final String error = "Unable to create summary";
+			log.error(error, e);
+			throw new ResponseStatusException(org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, error);
+		}
+	}
+
+	@PutMapping("/{id}")
+	@Secured(Roles.USER)
+	@Operation(summary = "Update a summary")
+	@ApiResponses(
+			value = {
+				@ApiResponse(
+						responseCode = "200",
+						description = "Summary updated.",
+						content =
+								@Content(
+										mediaType = MediaType.APPLICATION_JSON_VALUE,
+										schema =
+												@io.swagger.v3.oas.annotations.media.Schema(
+														implementation = Summary.class))),
+				@ApiResponse(responseCode = "404", description = "Summary could not be found", content = @Content),
+				@ApiResponse(
+						responseCode = "500",
+						description = "There was an issue updating the summary",
+						content = @Content)
+			})
+	public ResponseEntity<Summary> updateSummary(
+			@PathVariable("id") final UUID id,
+			@RequestBody final Summary summary,
```

Comment:
Do we need the `id` if the user is sending the full `Summary`? or is it custom to do so?

Reply:
its customary in our other endpoints